### PR TITLE
Options

### PR DESCRIPTION
**Pull request by g4s8**
#703 - options + `@todo`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#905](https://github.com/zerocracy/farm/issues/905) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#905](https://github.com/zerocracy/farm/issues/905) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/991215609); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/test/java/com/zerocracy/pmo/OptionsTest.java [hunk 1]
```diff
@@ -16,17 +16,87 @@
...
+                new TeeInput(
+                    new ResourceOf("com/zerocracy/pmo/options.xml"),
+                    item.path()
+                )
+            ).intValue();
+        }
+    }
+
+    @Test
+    public void readMaxJobs() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/OptionsTest.java]: @g4s8 let's add another test where `maxJobsInAgenda` can return its default value.

File: src/test/java/com/zerocracy/pmo/OptionsTest.java [hunk 2]
```diff
@@ -16,17 +16,87 @@
...
+        MatcherAssert.assertThat(
+            new Options(new Pmo(this.farm), OptionsTest.LOGIN)
+                .bootstrap()
+                .maxJobsInAgenda(0),
+            Matchers.equalTo(2)
+        );
+    }
+
+    @Test
+    public void readNotifyStudents() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/OptionsTest.java]: @g4s8 same here, let's have a test where the default value is returned.

File: src/test/java/com/zerocracy/pmo/OptionsTest.java [hunk 3]
```diff
@@ -16,17 +16,87 @@
...
+        MatcherAssert.assertThat(
+            new Options(new Pmo(this.farm), OptionsTest.LOGIN)
+                .bootstrap()
+                .notifyStudents(false),
+            Matchers.is(true)
+        );
+    }
+
+    @Test
+    public void readNotifyRfps() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/OptionsTest.java]: @g4s8 same here too, let's have another case where the default value is returned.

File: src/test/java/com/zerocracy/pmo/OptionsTest.java [hunk 4]
```diff
@@ -16,17 +16,87 @@
...
+        MatcherAssert.assertThat(
+            new Options(new Pmo(this.farm), OptionsTest.LOGIN)
+                .bootstrap()
+                .notifyRfps(false),
+            Matchers.is(true)
+        );
+    }
+
+    @Test
+    public void notifyPublish() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/OptionsTest.java]: @g4s8 ditto, let's have a test for returning the default value.

File: src/test/java/com/zerocracy/pmo/OptionsTest.java [hunk 5]
```diff
@@ -16,17 +16,87 @@
...
- *  name from `people.xml`) to read and modify user options.
- *  Also some mechanism should be invented to allow user to change this
- *  options.
  * @checkstyle JavadocMethodCheck (500 lines)
  */
 public final class OptionsTest {
+    /**
+     * User login.
+     */
+    private static final String LOGIN = "user1324234";
```
Comment (yegor256) [inline src/test/java/com/zerocracy/pmo/OptionsTest.java]: @g4s8 it's a bad practice, see: http://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html

File: src/test/java/com/zerocracy/pmo/OptionsTest.java [hunk 6]
```diff
@@ -16,17 +16,87 @@
...
+    @Before
+    public void setUp() throws Exception {
+        this.farm = new FkFarm();
+        try (
+            final Item item = new Pmo(this.farm)
+                .acq(String.format("options/%s.xml", OptionsTest.LOGIN))
+        ) {
+            new LengthOf(
+                new TeeInput(
+                    new ResourceOf("com/zerocracy/pmo/options.xml"),
```
Comment (yegor256) [inline src/test/java/com/zerocracy/pmo/OptionsTest.java]: @g4s8 same story: http://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html I would suggest to get rid of the file and use a private static method that builds the XML. We are not using files anywhere as of now, let's not start.

**Comment by yegor256** [comment]
@g4s8 see above

**Comment by g4s8** [comment]
@carlosmiranda @yegor256 fixed all comments in 7310716

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by g4s8** [comment]
@yegor256 ping

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by g4s8** [comment]
@yegor256 rultor failed to push, please restart merge
```
To git@github.com:zerocracy/farm.git
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'git@github.com:zerocracy/farm.git'
```

**Comment by yegor256** [comment]
@rultor merge

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/905#issuecomment-389008749))

@carlosmiranda This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#905](https://github.com/zerocracy/farm/issues/905) is [in scope](http://www.zerocracy.com/policy.html#14) for 1mon
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [1mon](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%23905%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by carlosmiranda** [comment]
@yegor256 can we close this job from WBS? Seems like 0crat missed this.

**Comment by carlosmiranda** [comment]
@yegor256 ping... can we close this job?

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/905#issuecomment-390187425))

@yegor256 The job [#905](https://github.com/zerocracy/farm/issues/905) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@g4s8 According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Pull request title explains the problem it is fixing;
Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Please add more details to PR's title and description.

**Comment by g4s8** [comment]
@ypshenychka right, updated title and added details to description, please look again.

**Comment by ypshenychka** [comment]
@g4s8 Thanks!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**